### DISCUSSION TITLE
ioBroker Integration

### DISCUSSION THREAD

**Discussion by FoggySolar** [solectrus/solectrus#3131] [Ideas]
Hallo,

ich hoffe meine Anfrage ist hier richtig? 
Noch bin ich recht frisch mit dem ioBroker habe aber meinem Senec v3 schon ordentlich implementieren können.

Nun meine spannende Frage: "Gibt es eine Möglichkeit die Daten von Solectrus in den ioBroker zu bekommen?"

Mir geht es im Endeffekt um folgendes.

Ich möchte gerne im ioBroker ein Script erstellen, welches mir (nicht im 5 Minuten sondern im 5 oder 10 Sekunden Intervall) Daten meiner **aktuellen** Einspeisung gibt, so dass ich bei zu viel Einspeisung meine Smart Steckdose zum Anschalten meiner Heizpatrone im Brauchwasser nutzen kann. 

Vielen Dank im Voraus

VG
 FoggySolar

**Comment by ledermann** [comment]
Klar bist du hier richtig :)

Du könntest auf die InfluxDB (von SOLECTRUS) von außen zugreifen und beliebige Flux-Abfragen durchführen, um dann ein Event auszulösen. In ioBroker ist ein InfluxDB-Client enthalten, du kannst damit also auf die SOLECTRUS-Datenbank zugreifen.

Oder, vielleicht noch besser: In InfluxDB (von SOLECTRUS) könntest du Alerts definieren, die bei bestimmten Messwerten Aktionen auslösen, z.B. einen HTTP-Request verschicken. Genaueres hier:
https://docs.influxdata.com/influxdb/v2/monitor-alert/

Um an das Frontend der InfluxDB heranzukommen, musst du in deiner `docker-compose.yml` den Port `8086` freigeben.

**Reply by FoggySolar** [reply]
Vielen Dank für deine Info.
Da werde ich einmal schauen ob ich das soweit hinbekomme, bin ja nicht der gelernte Programmierer.
VG
 FoggySolar

**Reply by FoggySolar** [reply]
Darf ich zu diesem Thema eine weitere Frage stellen? Wie gesagt ich bin noch vollkommener Anfänger auf dem Gebiet.
Im iobroker werde ich bei dem nun installierten InfluxDB Adapter nach folgenden Dingen gefragt:
_Organisation_, _Token_ und einem _DB-Name_.
In der docker-compose.yml habe ich den Port 8086 bereits freigeben mit dem entfernen von #.
Denke das ist korrekt. 
Aber wie bekomme ich die fehlenden Infos?
Vielleicht noch zur weiteren Info, installiert ist Solectrus auf einer Synology im Container. 
Vielen lieben Dank vorab für die tolle Hilfe.
VG
 FoggySolar

**Reply by FoggySolar** [reply]
OK, ich habe jetzt in der .env Datei einige Daten gefunden, entschuldige bitte.
Jedoch fehlt mir der DB-Name wahrscheinlich noch, der erschließt sich mir noch nicht, da der Adapter immer noch nicht glücklich gestellt ist im ioBroker.
VG
 FoggySolar

**Reply by ledermann** [reply]
Dass der ioBroker von "DB Name" spricht, ist etwas unglücklich, weil InfluxDB keine "Datenbanken" kennt, sondern dieses Konzept "Buckets" nennt. Was du also in deinem ioBroker eintragen musst, ist das, was in deiner `.env` unter `INFLUX_BUCKET` notiert ist.

Ansonsten hier noch der Hinweis, dass du nach der Änderung von `docker-compose.yml` (wegen Port 8086 für InflxuDB), die Container neu starten musst, damit das wirksam wird:

```
docker-compose down
docker-compose up -d
```

**Reply by FoggySolar** [reply]
Entschuldige bitte das ich mich jetzt erst melde, aber mein ioBroker hatte sich mit einem Update komplett verabschiedet.
Und als Neuling funktionierten die Backups nicht wie gewünscht.
So, die Verbindung zur InfluxDB steht, danke dir.
Nur und ich hoffe das ist meine letzte Frage dazu, habe ich jetzt die Verbindung zur DB aber wie komme ich jetzt an die Daten?
Als Objekt sehe ich aktuell nur im Baum: influxdb.0 -> Info -> connection = true
Muss ich im Adapter noch bei Standard- oder Experteneinstellungen noch Änderungen vornehmen?

**Reply by ledermann** [reply]
Wenn die Verbindung zu InfluxDB grundsätzlich funktioniert, dann könntest du nun mit der Erstellung deines Scriptes beginnen. Soweit ich das sehe, kann die [Script engine for JavaScript and Blockly](https://github.com/ioBroker/ioBroker.javascript) auch auf InfluxDB zugreifen. Genaueres weiß ich mangels Erfahrung damit aber nicht. Prinzipiell müsste es so gehen, dass du konkrete Flux-Queries bastelst, diese an InfluxDB schickst und das Ergebnis dann weiterverarbeitest. Wenn du nicht in InfluxDB direkt einloggst, kannst du über das dortige UI die von SOLECTRUS dort abgelegten Daten sehen und mit dem integrierten Query-Editor Abfragen erstellen.

Ich hoffe, ich habe dich nicht falsch verstanden. Ein solches Script muss von dir programmiert werden. Dass das einfach sei, habe ich nicht gesagt ;-)

**Reply by FoggySolar** [reply]
Vielen Dank für deine Erläuterungen, ich werde mal schauen ob ich das in die Reihe bekomme.
Zumindest ist der Anfang gemacht. Vielen Dank für deine Unterstützung. :)

**Reply by JuergenKir** [reply]
Bin leider eben erst auf dieses Thema gestoßen:
Ich nutze sowohl Solectrus (seit letzter Woche auch mit Wärmepumpe !! Danke an Georg @ledermann !!) als auch ioBroker und habe dafür 2 separate Container unter Proxmox. 
Der ioBroker hat bei mir eine eigene InfluxDB. Da ich dies schon eine Weile nutze ist das noch eine InfluxDB 1.x. 
Wenn ich dich richtig verstanden habe, hast du den Senec- und den influx-Adapter im ioBroker schon erfolgreich installiert. Dann kannst du einzelne Objekte aus dem Senec-Adaper in der influxDB mitschreiben. Der Senec-Adapter liefert die meisten Daten im 10 Sekunden Takt. Auswertungen und grafische Darstellungen mache ich dann mit Grafana. 
Für einfache Scripte eignet sich Blockly direkt im ioBroker. Um einen Heizstab bei Überschuss zu schalten brauchst du aber aus meiner Sicht dafür keine influxDB, sondern kannst direkt die Werte der Objekte im Script verwenden.

**Reply by FoggySolar** [reply]
Guten Morgen,
ja, aktuell nutze ich ioBroker in einer Synology als Container und Solectrus ebenfalls als separaten Container mit der dazugehörigen InfluxDB.
Mein Ziel ist es mit einem Blockly meinem per FritzDect 210 gesteckten Heizstab in meinem Brauchwasserpuffer zu steuern.
Das klappt per Blockly zur Zeit mit den Senec Daten, aber halt nur im ca. 5 Minuten Aktualisierungsintervall.
Daher möchte ich eben die Daten der InfluxDB gerne in den ioBroker liefern um so effizienter zu sein. ;)
Im InfluxDB stelle ich mich vermutlich zur Zeit einfach zu dusselig an um die Daten zu bekommen.
Die Daten in der InfluxDb sehe ich zwar, aber ich weiß einfach, da das ganze Thema für mich absolutes Neuland ist, nicht wo und wie ich die Daten in den ioBroker bekomme.
In der InfluxDB unter DataExplorer sehe ich ja "solectrus" -> SENEC -> und dann alle Daten.
Ich dachte auch schon das ich eine neue "Buckets" anlegen muss, aber irgendwie stecke ich fest.
Vorab bin ich dir schon einmal sehr dankbar für deine Hilfe.

**Reply by JuergenKir** [reply]
Hallo,
Der Senec-Adapter liefert die meisten Daten bei mir alle 10 Sekunden. Dafür habe ich im Senec-Adapter unter den Allgemeinen Einstellungen "Verbinden über lala.cgi" aktiviert. Dadurch wird direkt auf die Senec-Daten im lokalen Netzwerk zugegriffen anstatt auf den Senec-Server. High Prio Polling habe ich auf 10 Sekunden gestellt. Low Prio Polling auf 10 Minuten. Zusätzlich muss HTTPS aktiviert werden. In der aktuellen Version des Senec-Adapters kann mittlerweile sogar sehr detailliert eingestellt werden, welche Daten High Prio oder Low Prio sind.
Der entscheidende Wert für dein Script wäre aus meiner Sicht "senec.0.ENERGY.GUI_GRID_POW". Das ist die Netzeinspeisung bzw. Netzbezug.
Diesen Wert kannst du dann direkt im Blockly-Script verwenden; ohne Umweg über InfluxDB.

**Reply by FoggySolar** [reply]
Hallo,
du bist ja SUPER, das ist genau das was ich gesucht habe. 
Darauf wäre ich im Leben nicht gekommen, das unter SENEC ENERGY zu suchen.
Ich habe immer die Daten unter SENEC API Anlage ... gezogen.
Da sind all meine Probleme sofort geklärt und ich muss mein Blockly nur noch dahingehend anpassen.
Spitze, du und Georg @ledermann, ihr wisst gar nicht wie glücklich ich gerade bin.
So wird der SENEC Kasten doch noch zu etwas gut mit all seinen aktuellen Problemen.
Vielen vielen vielen DANK an euch beide für diese tolle Unterstützung!
